File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
   void solve(vector<int>&nums,vector<int>&output,vector<vector<int>>&ans,int index,int n){
        
            ans.push_back(output);
            
        for(int i=index;i<n;i++){
            if(i>index && nums[i]==nums[i-1])continue;
            output.push_back(nums[i]);
            solve(nums,output,ans,i+1,n);
            output.pop_back();
        }
   }

    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        vector<int>output;
        vector<vector<int>>ans;
        solve(nums,output,ans,0,n);
        return ans;
    }
};

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    public  static void solve(int[] nums,List<Integer>output,List<List<Integer>>ans,int ind){
        
            ans.add(new ArrayList<>(output));
           
        for(int i=ind;i<nums.length;i++){
            if(i>ind && nums[i]==nums[i-1])continue;
            output.add(nums[i]);
            solve(nums,output,ans,i+1);
            output.remove(output.size()-1);
        }
    }

    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<Integer>output=new ArrayList<>();
        List<List<Integer>> ans=new ArrayList<>();
        solve(nums,output,ans,0);
        return ans;
    }
}